opm/autodiff/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -195,7 +195,9 @@ namespace Opm
             data::Wells res = WellState::report();
 
             const int nw = this->numWells();
-            const int np = this->numPhases();
+            // If there are now wells numPhases throws a floating point
+            // exception.
+            const int np = nw ? this->numPhases() : -1;
```

Comment:
A more elegant fix: move the np variable inside the for-loop, since it is only used there. (Won't require, tell me if you want to do that or keep this as currently proposed.)